README.md

```diff
@@ -109,6 +109,25 @@ See [action.yml](./action.yml) for a full list of supported arguments and their
     fail: true
 ```
 
+## Utilising the cache feature
+
+In order to mitigate issues regarding rate limiting or to reduce stress on external resources, one can setup lychee's cache similar to this:
+
+```yml
+- name: Restore lychee cache
+  uses: actions/cache@v3
+  with:
+    path: .lycheecache
+    key: cache-lychee-${{ github.sha }}
+    restore-keys: cache-lychee-
```

Comment:
>Should we add another step at the end which stores the cache for the next run? I'd assume that's what most people would want to do.

There will automatically be a `Post` step (generated from the used [actions/cache](https://github.com/actions/cache) Action) taking care of that.

It will compare and save the cache based on the given key.
So in this setup, as long as the user triggers the CI run from the same commit, it will be the same key. The first run will save the cache, subsequent runs will not update it (same commit hash).
For continuous cache updates on any run, the only solution that I came up with was using a timestamp. But that would require adding an extra step just to get it, there is no variable available.

For restoring the cache, the most recent available one is used (commit hash doesn't matter).

Reply:
Added a note as discussed.